include/kframework/lemmas/int-simplification.k

```diff
@@ -158,4 +161,43 @@ module INT-SIMPLIFICATION-COMMON
 
     rule A -Int B +Int C <=Int D => false requires D <Int A -Int B andBool 0 <=Int C [simplification]
 
+  // ###########################################################################
+  // up/Int
+  // ###########################################################################
+
+    rule [upInt-lt-true]:
+      ((X up/Int Y) *Int Y) <Int Z => true
+      requires X +Int Y <=Int Z andBool 0 <Int Y
+      [simplification]
+
+    rule [upInt-lt-false]:
+      ((X up/Int Y) *Int Y) <Int Z => false
+      requires Z <=Int X andBool 0 <Int Y
+      [simplification]
+
+    rule [upInt-refl-leq]:
+      ((X up/Int Y) *Int Y) <=Int X => X modInt Y ==Int 0
+      requires 0 <Int Y
+      [simplification]
+
+    rule [upInt-refl-gt]:
+      X <Int ((X up/Int Y) *Int Y) => X modInt Y =/=Int 0
+      requires 0 <Int Y
```

Comment:
We should really pull these out into their own module, `EVM-WORD-SIMPLIFICATION` perhaps.

That's because `up/Int` is not part of the prelude, and I intend for `INT-SIMPLIFICATION` to eventually be upstreamed to the K prelude so that all K definitions benefit from it.